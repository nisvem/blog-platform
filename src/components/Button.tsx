import React from 'react';

interface IButton extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  children?: React.ReactNode;
}

export const Button: React.FC<IButton> = ({ children, ...attr }) => {
  return (
    <button
      className="block mx-auto bg-accent text-accent-text px-4 py-2 rounded-xl mt-4 hover:cursor-pointer hover:opacity-90 hover:scale-[1.05] transition-all w-full "
      {...attr}
    >
      {children}
    </button>
  );
};