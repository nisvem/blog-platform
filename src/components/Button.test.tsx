import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { Button } from './Button';
import '@testing-library/jest-dom';

describe('Button component', () => {
  it('renders with provided text', () => {
    render(<Button>Click me</Button>);
    expect(screen.getByRole('button')).toHaveTextContent('Click me');
  });

  it('calls onClick handler when clicked', () => {
    const handleClick = jest.fn();
    render(<Button onClick={handleClick}>Click me</Button>);
    fireEvent.click(screen.getByRole('button'));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(<Button>Click me</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
}); 