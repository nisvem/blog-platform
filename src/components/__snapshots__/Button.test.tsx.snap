// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches the snapshot 1`] = `
<DocumentFragment>
  <button
    class="block mx-auto bg-accent text-accent-text px-4 py-2 rounded-xl mt-4 hover:cursor-pointer hover:opacity-90 hover:scale-[1.05] transition-all w-full "
  >
    Click me
  </button>
</DocumentFragment>
`;
